examples/dbscan/converter.cpp

```diff
@@ -0,0 +1,445 @@
+  {
+    if (lineNo > -1)
+    {
+      char *pch;
+      char *end_str;
+      wordNo = 0;
+      lonlatno = 0;
+      pch = strtok_r(line, "\"[", &end_str);
+      while (pch != nullptr)
+      {
+        if (wordNo > 0)
+        {
+          char *pch2;
+          char *end_str2;
+
+          pch2 = strtok_r(pch, ",", &end_str2);
+
+          if (strcmp(pch2, "]") < 0 && lonlatno < 255)
+          {
+
+            thisWord = atof(pch2);
+
+            if (thisWord != 0.00000)
+            {
+              if (thisWord > -9 && thisWord < -7)
+              {
+                longitudes.push_back(thisWord);
+                // printf("lon %f",thisWord);
+                pch2 = strtok_r(nullptr, ",", &end_str2);
+                thisWord = atof(pch2);
+                if (thisWord < 42 && thisWord > 40)
+                {
+                  latitudes.push_back(thisWord);
+                  // printf(" lat %f\n",thisWord);
+
+                  lonlatno++;
+                }
+                else
+                {
+                  longitudes.pop_back();
+                }
+              }
+            }
+          }
+        }
+        pch = strtok_r(nullptr, "[", &end_str);
+        wordNo++;
+      }
+      // printf("num lonlat were %d x 2\n",lonlatno);
+    }
+    lineNo++;
+    if (lonlatno <= 0)
+    {
+      lineNo--;
+    }
+
+    // printf("Line %d\n",lineNo);
+  }
+  fclose(fp_data);
+
+  int num_points = longitudes.size();
+  assert(longitudes.size() == latitudes.size());
+  v.reserve(num_points);
+  for (int i = 0; i < num_points; ++i)
+    v.emplace_back(longitudes[i], latitudes[i], 0.f);
+
+  std::cout << "done\nRead in " << v.size() << " points" << std::endl;
+
+  return v;
+}
+
+// 3D Road Network data reader.
+//
+// The data consists of more than 400,000 points from the road network of North
+// Jutland in Denmark.
+//
+// The data can be found at
+// https://archive.ics.uci.edu/ml/datasets/3D+Road+Network+(North+Jutland,+Denmark)
+// (direct link
+// https://archive.ics.uci.edu/ml/machine-learning-databases/00246/3D_spatial_network.txt).
+//
+// Each data point contains its ID, longitude, latitude, and altitude.
+std::vector<Point> load3DRoadNetworkData(std::string const &filename)
+{
+  std::cout << "Assuming 3DRoadNetwork data.\n";
+  std::cout << "Reading in \"" << filename << "\" in text mode...";
+  std::cout.flush();
+
+  std::ifstream file(filename);
+  assert(file.good());
+  if (!file.good())
+    throw std::runtime_error("Cannot open file");
+
+  std::vector<Point> v;
+
+  std::string thisWord;
+  while (file.good())
+  {
+    getline(file, thisWord, ',');
+    getline(file, thisWord, ',');
+    float longitude = stof(thisWord);
+    getline(file, thisWord, ',');
+    float latitude = stof(thisWord);
+    v.emplace_back(longitude, latitude, 0.f);
+  }
+  // In Mustafa2019 they discarded the last item read but it's not quite clear
+  // if/why this was necessary.
+  // lon_ptr.pop_back();
+  // lat_ptr.pop_back();
```

Comment:
I guess we can remove comments like this and similar now.

Reply:
No, we should leave this in, it is helpful to have that information.